packages/devextreme/js/__internal/grids/grid_core/keyboard_navigation/m_keyboard_navigation.ts

```diff
@@ -1449,14 +1450,19 @@ export class KeyboardNavigationController extends KeyboardNavigationControllerCo
         const $target = event
           && $(event.target).closest(`${NON_FOCUSABLE_ELEMENTS_SELECTOR}, td`);
         const skipFocusEvent = $target && $target.not($cell).is(NON_FOCUSABLE_ELEMENTS_SELECTOR);
-        const isEditor = !!column && !column.command && $cell.hasClass(EDITOR_CELL_CLASS);
-        const isDisabled = !isEditor && (!args.isHighlighted || skipFocusEvent);
+        const isEditCell = !!column
+          && !column.command
+          && isEditing
+          && $cell.hasClass(EDITOR_CELL_CLASS);
+        const hasEditorAlways = !isEditing && column?.showEditorAlways;
+        const isDisabled = (!isEditCell || hasEditorAlways)
```

Comment:
We have a redundant term here.
(!isEditCell || hasEditorAlways) === isEditCell.

```suggestion
        const isDisabled = !isEditCell && (!args.isHighlighted || skipFocusEvent);
```